presto-main/src/main/java/com/facebook/presto/spiller/SingleStreamSpiller.java

```diff
@@ -46,6 +48,11 @@
     Iterator<Page> getSpilledPages();
 
     /**
+     * Initiates read of previously spilled pages. The returned {@link Future} will be complete once all pages are read.
+     */
+    ListenableFuture<List<Page>> getAllSpilledPages();
```

Comment:
Do we care about how much memory the "unspilled" pages would consume.
Maybe the `SSS` should be extended with `getSpilledSize()` method?